docs/advanced-features/event-driven-packaging.md

```diff
@@ -0,0 +1,177 @@
+Sync](https://aws.amazon.com/datasync/) that write objects to Amazon S3.
+
+> Any AWS service or action that generates S3 object events may trigger the EDP service.
+
+## Requirements
+
+1. A pre-existing VPC that includes a [NAT
+Gateway](https://docs.aws.amazon.com/vpc/latest/userguide/vpc-nat-gateway.html)
+or the following [VPC
+endpoints](https://docs.aws.amazon.com/vpc/latest/privatelink/concepts.html#concepts-vpc-endpoints):
+  - Amazon S3
+  ([gateway](https://docs.aws.amazon.com/vpc/latest/privatelink/gateway-endpoints.html)
+  or
+  [interface](https://docs.aws.amazon.com/vpc/latest/privatelink/create-interface-endpoint.html))
+  - EventBridge (interface)
+2. [Enable EventBridge S3
+Events](https://docs.aws.amazon.com/AmazonS3/latest/userguide/enable-event-notifications-eventbridge.html)
+for all S3 buckets to be monitored by EDP
+
+![](../imgs/edp-event-bridge-s3-properties.png)
+
+## Deployment
+
+To use EDP in your Quilt instance, you will use a standalone
+CloudFormation template to configure when, and how, events get
+generated.
+
+### Parameters
+
+The following parameters need to be defined in the CloudFormation
+template during deployment:
+
+- `VPC`: Same as the existing Quilt instance.
+- `SecurityGroup`: Same as the existing Quilt instance.
+- `Subnets`: Same two subnets as the existing Quilt instance.
+- `BucketName`: Name of the Amazon S3 bucket to monitor.
+- `BucketIgnorePrefixes`: List of bucket path segments to ignore
+- `BucketPrefixDepth`: The number of `/`-separated *common* path segments
+at the beginning of an S3 object key. Default value is `2`.
+- `BucketThresholdDuration`: Trigger a notification when this number
+of seconds has elapsed since the last object event in the S3 bucket 
+occurred. Default value is `300` seconds.
+- `BucketThresholdEventCount`: Trigger a notification when this
+number of files have been created (since the prior trigger)
+- `DBUser`: Username for EDP RDS instance
+- `DBPassword`: Password for EDP RDS instance
+- `EventBusName`: Name of custom EventBridge event bus that recieves events
+
+![](../imgs/edp-cloudformation-parameters.png)
+
+## How EDP works
+
+1. EDP monitors S3 object events for _s3://source-bucket_
+2. After a fixed number of events (`BucketThresholdEventCount`) or
+a maximum duration within a common prefix (`BucketThresholdDuration`) EDP creates a
+package in _s3://target-bucket_ (the _target_ bucket may or may not be
+the same as the _source_ bucket).
+3. Each EDP event contains sufficient information for the
+recipient to make Quilt data packages from the event:
+  - S3 Bucket name
+  - Common prefix
+  - Number of files
+  - Timestamp of event
+4. EDP ignores changes to ignored prefixes (`BucketIgnorePrefixes` parameter)
+in the source S3 bucket (e.g. `raw/*`)
+5. EDP publishes events to an AWS EventBridge bus and can forward
+events to external EventBridge buses
+6. Users subscribe the Amazon SNS topic created to recieve notifications
+
+## Example EDP events
```

Comment:
This section and the docs as a whole are missing the all-important example of a rule (in code, not a screenshot) to catch EDP events and forward them to a target. That’s how people will actually use EDP and should be the most salient part of this document. Installation is mostly one-time whereas EB rules and targets are the common place where EDP gets used.

Also you need to explain limits, such as monitors one bucket at present …

Reply:
Also need to mention stack has an SNS topic (useful for debugging and seeing how events are structured)

Reply:
The key thing about the short code snippet for the rule is not to redo the Amazon docs on rules but to show how to match EDP events. Something like “[com.quilt data.edp]” is almost there